backend/src/database/migrations/20221117031825-update-insert-job.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.createFunction(
+      'insert_job',
+      [
+        { type: 'integer', name: 'selected_campaign_id' },
+        { type: 'integer', name: 'selected_send_rate' },
+        { type: Sequelize.DataTypes.STRING(255), name: 'scheduled_timing' },
+        { type: 'integer', direction: 'OUT', name: 'selected_job_id' },
+      ],
+      'integer',
+      'plpgsql',
+      `
+INSERT INTO job_queue ("campaign_id", "send_rate", "status", "created_at", "updated_at", "visible_at")
+VALUES (selected_campaign_id, selected_send_rate, 'READY', clock_timestamp(), clock_timestamp(), scheduled_timing::timestamp)
+RETURNING id INTO selected_job_id;
+`,
+      [],
+      {
+        force: true,
+      }
+    )
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.dropFunction('insert_job', [
```

Comment:
Actually, more generally, why don't we use a single migration instead of splitting them across four different files? Since I presume they're all or nothing and are not meant to be incrementally deployed?

Reply:
Pros and cons of single migration.

Pros:
- Easy to see all the db changes needed for scheduled sending to work in a single file
- Easier to revert (just run once)

Cons:
- If a specific step in the migration goes wrong, harder to undo (need to comment out the other un-related down steps)

Reply:
@tan-zx @richard-yak just wanna give my 2 cent, usually I do 1 migration max per PR. For this one @tan-zx , I think part of the reason why we have multiple migration is because this is an aggregate of multiple PRs

Reply:
> @tan-zx @richard-yak just wanna give my 2 cent, usually I do 1 migration max per PR. For this one @tan-zx , I think part of the reason why we have multiple migration is because this is an aggregate of multiple PRs

Occurred to me that, if we have a single migration, we can wrap the different steps in a single transaction, so that if any part fails, the whole thing can automatically rollback 🤔

also, forgot to note: think in my chat with Richard ytd:
- we are keeping the old `insert_job` so that we can have backwards compatibility
- we will the new `insert_job` differently so that it's clearer which is meant to run at any one point

Reply:
Tackle in a separate issue: https://github.com/opengovsg/postmangovsg/issues/1893